src/user.js

```diff
@@ -21,22 +23,42 @@ class User {
 	}
 	
   searchForRecipe(keyword, array) {
+    let byName = this.searchByName(keyword, array)
+    let byIngredient = this.searchByIngred(keyword, array)
+    let searchedResults = byName.concat(byIngredient)
+    console.log('results', searchedResults)
+    return [... new Set(searchedResults)]
+  }
+
+  searchByName(keyword, array) {
     let searchedResults = []
     array.forEach(recipe => {
-      if(recipe.name.includes(keyword)) {
+      if(recipe.name.toLowerCase().includes(keyword)) {
         searchedResults.push(recipe)
       }
-      console.log('recipeIngred', recipe.ingredients)
+    })
+    console.log('name', searchedResults)
+    return searchedResults
+  }
+//s&p issue
+  searchByIngred(keyword, array) {
+    let searchedResults = []
+    let ingredientIDs = ingredientsData
+      .filter(ingred => ingred.name)
+      .filter(ingred => ingred.name.toLowerCase().includes(keyword))
+      .map(ingred => ingred.id)
```

Comment:
is line 47 necessary? seems like it's adding a 3rd loop that isn't doing any conditional check

Reply:
Line 47 is filtering out the ones that have a name attached.  the dataset that i was testing through had an item without a name property so it was breaking everything.  we can get rid of it if we alter the data.